app/gui2/src/util/ast/__tests__/__snapshots__/raw.test.ts.snap

```diff
@@ -50,6 +50,7 @@ exports[`Parsing '2
                 "spanLeftOffsetCodeStartLine": 1,
                 "spanLeftOffsetCodeStartUtf8": 4,
                 "type": "Number",
+                "warnings": [],
```

Comment:
Do I read this correctly that `warnings` are propagated to TypeScript. How are the warnings propagated to CLI? I don't see any changes in `Tree`. How will the CLI report these warnings?

Reply:
There is `Tree.getWarnings`, however they are serialized as warning IDs (plus warning-specific details), with a separate table providing the string/template to display each warning ID. I have not yet implemented the JNI to turn this in to a `getMessage` for the `Warning` object.

Reply:
In order to expose the linter via `enso-engine*/bin/enso` we will need a way to access the warnings from JVM.

Reply:
Yes, expect a PR soon. I just wanted to get this part merged since it touches a lot of files.